### ISSUE TITLE
pihole-FTL startup fails with: FATAL: create_shm(): Failed to resize "FTL-lock" (4) to 28: Operation not supported (-1)

### ISSUE THREAD

**Issue by trupf** [pi-hole/pi-hole#3921]
### Versions

- Pi-hole:  v5.2.1
- AdminLTE:  v5.2.1
- FTL:  v5.3.2

### Platform

- OS and version:  Debian 10

- Platform:  armhf

### Expected behavior

pihole-FTL should start (and it did until I updated to the latest version)

### Actual behavior / bug

start of pihole-FTL fails with the above error message

### Steps to reproduce

just give the command:  
sudo -u /usr/bin/pihole-FTL -f

## Debug Token

- URL:  https://tricorder.pi-hole.net/wim5hft4rq

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Additional context

_Add any other context about the problem here._

**Comment by dschaper** [comment]
And I need your actual debug token, the one in your post is the default link from the template.

**Comment by trupf** [comment]
> > sudo -u /usr/bin/pihole-FTL -f
> 
> Why are you running that command?

I'm just using it to see the output (error message). as it is easier than every time starting the service and checking the log file. 
Usually i start the service with "service pihole-FTL start". But that command fails as well and I can find the same error message in the log.

The debug token should be the correct one, I have just mistakenly replaced the token in the template as well. If not I would have to upload a debug file again. 

Also I have now replaced the pihole-FTL binary with an older one, version 5.2 is the last one running fine on my system, beginning with version 5.3 I get the above error.

**Comment by dschaper** [comment]
Running `sudo -u /usr/bin/pihole-FTL -f` will start `pihole-FTL` as root and will change the ownership of files (like the SHM). Further running of Pi-hole with `service` or `systemctl` will fail as the `pihole` user can not read/write the SHM files created by root when you ran it as `sudo`.

**Comment by dschaper** [comment]
https://tricorder.pi-hole.net/wim5hft4rq doesn't exist, that's the example URL.

**Comment by dschaper** [comment]
And that appears to be the problem:

```
*** [ DIAGNOSING ]: contents of /dev/shm
-rw------- 1 root root 0 Dez 12 18:15 /dev/shm/FTL-lock
```

You need to remove that file, and don't run `pihole-FTL` directly as root or things will break. This is a proper directory:

```plain
drwxr-xr-x 18 root   root      3620 Nov 29 15:17 ..
-rw-------  1 pihole pihole  339968 Dec  7 13:46 FTL-clients
-rw-------  1 pihole pihole     152 Dec  7 13:46 FTL-counters
-rw-------  1 pihole pihole  229376 Dec  7 13:46 FTL-dns-cache
-rw-------  1 pihole pihole  131072 Dec  7 13:46 FTL-domains
-rw-------  1 pihole pihole      28 Dec 12 11:17 FTL-lock
-rw-------  1 pihole pihole   20480 Dec  7 13:46 FTL-overTime
-rw-------  1 pihole pihole    4096 Dec  7 13:46 FTL-per-client-regex
-rw-------  1 pihole pihole 7110656 Dec  7 13:59 FTL-queries
-rw-------  1 pihole pihole      12 Dec  7 13:46 FTL-settings
-rw-------  1 pihole pihole  221184 Dec 12 10:39 FTL-strings
-rw-------  1 pihole pihole   28672 Dec  7 13:46 FTL-upstreams
```

**Comment by trupf** [comment]
the new debug token is  https://tricorder.pi-hole.net/maxa98y4c2

**Comment by dschaper** [comment]
There have been a handful of unsupported changes and configurations. 

If you are Experiencing issues with a Pi-hole install that has non-standard elements (e.g you are using nginx instead of lighttpd, or there is some other aspect of your install that is customised) - please use the [Community Help](https://discourse.pi-hole.net/c/bugs-problems-issues/community-help/36) category.

**Comment by trupf** [comment]
Yes I use nginx, but that is just the web frontend - and I have no problem with it. The problem is that the resolver itself does not even start  And no, removing /dev/shm/FTL-lock and starting pihole-FTL as service didn't help.

**Comment by dschaper** [comment]
New debug token after you removed the root owned files and have tried starting Pi-hole via the `service|systemctl` commands please.

**Comment by trupf** [comment]
https://tricorder.pi-hole.net/xd89fqrjaa

**Comment by dschaper** [comment]
What hardware is this being run on? Is this a virtualized environment? 

What is the output of `ls -la /dev/shm`?

Please provide the output from:

`cat /etc/init.d/pihole-FTL`

Did you make your own `pihole-FTL.service` file? 

`sudo systemctl cat pihole-FTL`

**Comment by trupf** [comment]
I have used both variants, your init file and my service file, both give the same result (even after removing files in /dev/shm).


not virtulized ist a Zyxel NAS 
uname -a:  Linux nas 3.2.0-6-nas5xx #50984 SMP Debian 3.2.102-1 armv7l GNU/Linux

root@nas:/var/log# ls -la /dev/shm
insgesamt 0
drwxrwxrwt  2 root   root     60 Dez 12 22:09 .
drwxr-xr-x 13 root   root   2260 Dez 12 18:30 ..
-rw-------  1 pihole pihole    0 Dez 12 22:09 FTL-lock


root@nas:/Temp# cat /etc/init.d/pihole-FTL
```
#!/usr/bin/env bash
### BEGIN INIT INFO
# Provides:          pihole-FTL
# Required-Start:    $remote_fs $syslog $network
# Required-Stop:     $remote_fs $syslog $network
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: pihole-FTL daemon
# Description:       Enable service provided by pihole-FTL daemon
### END INIT INFO

FTLUSER=pihole
PIDFILE=/run/pihole-FTL.pid

is_running() {
    pgrep -o "pihole-FTL" > /dev/null 2>&1
}


# Start the service
start() {
  if is_running; then
    echo "pihole-FTL is already running"
  else
    # Touch files to ensure they exist (create if non-existing, preserve if existing)
    touch /var/log/pihole-FTL.log /var/log/pihole.log
    touch /tmp/var/log/pihole-FTL.log /tmp/var/log/pihole.log
    touch /run/pihole-FTL.pid /run/pihole-FTL.port
    touch /etc/pihole/dhcp.leases
    mkdir -p /run/pihole
    mkdir -p /var/log/pihole
    chown pihole:pihole /run/pihole /var/log/pihole
    # Remove possible leftovers from previous pihole-FTL processes
    rm -f /dev/shm/FTL-* 2> /dev/null
    rm /run/pihole/FTL.sock 2> /dev/null
    # Ensure that permissions are set so that pihole-FTL can edit all necessary files
    chown pihole:pihole /run/pihole-FTL.pid /run/pihole-FTL.port
    chown pihole:pihole /etc/pihole /etc/pihole/dhcp.leases 2> /dev/null
    chown pihole:pihole /var/log/pihole-FTL.log /var/log/pihole.log
    chown pihole:pihole /tmp/var/log/pihole-FTL.log /tmp//var/log/pihole.log
    chmod 0644 /var/log/pihole-FTL.log /tmp/var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log /tmp/var/log/pihole.log
    # Chown database files to the user FTL runs as. We ignore errors as the files may not (yet) exist
    chown pihole:pihole /etc/pihole/pihole-FTL.db /etc/pihole/gravity.db 2> /dev/null
    if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE+eip "$(which pihole-FTL)"; then
      su -s /bin/sh -c "/usr/bin/pihole-FTL" "$FTLUSER"
    else
      echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
      pihole-FTL
    fi
    echo
  fi
}

# Stop the service
stop() {
  if is_running; then
    pkill -o pihole-FTL
    for i in {1..5}; do
      if ! is_running; then
        break
      fi

      echo -n "."
      sleep 1
    done
    echo

    if is_running; then
      echo "Not stopped; may still be shutting down or shutdown may have failed, killing now"
      pkill -o -9 pihole-FTL
      exit 1
    else
      echo "Stopped"
    fi
  else
    echo "Not running"
  fi
  echo
}

# Indicate the service status
status() {
  if is_running; then
    echo "[ ok ] pihole-FTL is running"
    exit 0
  else
    echo "[    ] pihole-FTL is not running"
    exit 1
  fi
}


### main logic ###
case "$1" in
  stop)
        stop
        ;;
  status)
        status
        ;;
  start|restart|reload|condrestart)
        stop
        start
        ;;
  *)
        echo $"Usage: $0 {start|stop|restart|reload|status}"
        exit 1
esac

exit 0
```

Systemd service file (currently deactivated.)
root@nas:/Temp# cat /etc/systemd/system/pihole-FTL.service.save 
```
[Unit]
Description=Pi-hole FTLDNS
; This unit is supposed to indicate when network functionality is available, but it is only
; very weakly defined what that is supposed to mean, with one exception: at shutdown, a unit
; that is ordered after network.target will be stopped before the network
After=network.target
; A target that should be used as synchronization point for all host/network name service lookups.
; All services for which the availability of full host/network name resolution is essential should
; be ordered after this target, but not pull it in.
Wants=nss-lookup.target
Before=nss-lookup.target
;ConditionCapability=CAP_NET_BIND_SERVICE
;ConditionCapability=CAP_NET_RAW 
;ConditionCapability=CAP_NET_ADMIN

[Service]
Type=forking
Restart=on-abnormal
User=pihole
Group=pihole
;NoNewPrivileges=true
;PrivateTmp=true
ProtectHome=yes
#AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_NET_ADMIN CAP_SYS_NICE
#CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_NET_ADMIN CAP_SYS_NICE
ExecStartPre=/bin/bash -c '/sbin/setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE+eip /usr/bin/pihole-FTL'
PermissionsStartOnly=true

PIDFile=/run/pihole-FTL.pid

ExecStartPre=/bin/touch /var/log/pihole-FTL.log /tmp/var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log /tmp/var/log/pihole.log
ExecStartPre=/bin/mkdir -p /var/run/pihole
ExecStartPre=/bin/chown pihole:pihole /var/run/pihole
ExecStartPre=-/bin/rm -f /dev/shm/FTL-*
#ExecStartPre=-/bin/rm /run/pihole/FTL.sock
ExecStartPre=-/bin/rm -rf /var/run/pihole/FTL.sock
ExecStartPre=/bin/chown pihole:pihole /var/log/pihole-FTL.log /tmp/var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /etc/pihole /etc/pihole/dhcp.leases /var/log/pihole.log /tmp/var/log/pihole.log
ExecStartPre=/bin/chmod 0644 /var/log/pihole-FTL.log /tmp/var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
ExecStartPre=/bin/echo "nameserver 127.0.0.1" | /sbin/resolvconf -a lo.piholeFTL

ExecStart=/usr/bin/pihole-FTL
RestartSec=30s
ExecReload=/bin/kill -HUP $MAINPID

; Use graceful shutdown with a reasonable timeout
TimeoutStopSec=10s

; Make /usr, /boot, /etc and possibly some more folders read-only...
ProtectSystem=full
; ... except /etc/pihole
; This merely retains r/w access rights, it does not add any new.
; Must still be writable on the host!
ReadWriteDirectories=/etc/pihole

[Install]
WantedBy=multi-user.target
```

**Comment by dschaper** [comment]
You `.service` file is causing problems.

Again, you've made a number of changes to things and are running things in ways we don't provide support for. 

If you are Experiencing issues with a Pi-hole install that has non-standard elements (e.g you are using nginx instead of lighttpd, or there is some other aspect of your install that is customised) - please use the [Community Help](https://discourse.pi-hole.net/c/bugs-problems-issues/community-help/36) category.

**Comment by trupf** [comment]
As I wrote I have removed the service file already before I wrote here but it didn't help. It was neccessary when I installed pihole to get it running properly (And I think a service file should be included in your distribution anyway). And again the point is, version 5.2 of pihole-FTL is running fine with all files created in /dev/shm, version 5.3 and above does not create any. So I don't think it has anything to do with my setup. I will use the older version 5.2 for now, may be a later update will work again. 
I doubt Community can help me here.
